File: app/CausalityCSharp/CausalityLibrary/Type/Cause.cs
﻿using System;

namespace CausalityLibrary.Type
{
    [Serializable]
    public sealed class Cause
    {
        public SerialNumber SerialNumber
        { get; }

        public string Description
        { get; }

        public SerialNumber Perceptron
        { get; }

        public readonly SerialNumber[] Next;
        public readonly double[] Thresholds;

        public Cause(SerialNumber serialNumber,
            string description, SerialNumber perceptron,
            SerialNumber[] next, double[] thresholds)
        {
            SerialNumber = serialNumber;
            Description = description;
            Perceptron = perceptron;
            Next = next;
            Thresholds = thresholds;
        }
    }
}

File: app/CausalityJava/src/com/inha/dsp/causality/type/Cause.java
package com.inha.dsp.causality.type;

import java.io.Serializable;

public class Cause implements Serializable {
    private SerialNumber serialNumber;
    private String description;
    private SerialNumber perceptron;

    public SerialNumber getSerialNumber() {
        return serialNumber;
    }
    public String getDescription() {
        return description;
    }
    public SerialNumber getPerceptron() {
        return perceptron;
    }

    public final SerialNumber[] Next;
    public final double[] Thresholds;

    public Cause(SerialNumber sn, String desc, SerialNumber perc,
                 SerialNumber[] next, double[] thresh)
    {
        serialNumber = sn;
        description = desc;
        perceptron = perc;
        Next = next;
        Thresholds = thresh;
    }
}
